adr/data-test.md

```diff
@@ -0,0 +1,55 @@
+# [data-test] attributes
+
+## Context
+
+Quality Assurance teams perform automated tests on Products for non-regression testing.
+
+## Problems
+
+QA teams lean on HTML ID attributes or XPath to automate tests on UI, and updates on the shared library can break these scenarios.
+
+## Solutions
+
+Expose [data-test] attributes for QA purpose suggest in the components or layout markup.
+1 — By documenting them, the first benefit will be to act on a contract with our QA team.
+2 — By using them, after all, the second benefit will also be to perform component testing.
+
+```css
+[data-test="<block_name>.<element_type>[?<element_index>].<?element_identifier>"]
+```
+
+| Identifier           | Optional | Description                                                                                                                                                      |
+| -------------------- | -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `block_name`         |          | Component or layout identifier from our design language (ex: modal, search, password, inlineediting, etc.)                                                       |
+| `element_type`       |          | Element or its type used (ex: button, link, input, textarea, radio, etc.)                                                                                        |
+| `element_index`      | yes      | Element index if it's repeated (radio, menu items, etc.)                                                                                                         |
+| `element_identifier` | yes      | A short and comprehensive identifier when we talk about it each other, in case of a form field we can reuse its label value (reveal, cancel, edit, submit, etc.) |
+
+We need to precise something,
+
+> To decide which values to use and to live in the Design System, we definitively have to be **context agnostic**.
+> It seems we can't determine if we will use the input for a `first name` or an `API key name`.
+> That's why we need to **keep it simple and stupid**.
```

Comment:
```suggestion
Stick to this ruleset but keep in mind:

> This ruleset applies to Design System components: we definitively have to be **context agnostic**.
> If we're testing a text input, it could for instance be used for a `first name` or an `API key name`.
> That's why we need to **keep it simple and stupid** when naming things.

Describe the pattern, not the specific one-shot use you will use it for.
```